lib/reducers/recentBlocks.js

```diff
@@ -23,17 +23,24 @@ function recentBlocksReducer(state = initialState, action) {
       const block = payload;
       const { numBlocks } = newState;
 
-      // skip if we already have the block for that height
-      // or recentBlocks haven't been hydrated yet
-      // reason is new block can be received multiple times
-      if (
-        (blocks && blocks.length && !blocks[block.height]) ||
-        blocks[block.height].hash !== block.hash
-      ) {
-        blocks = [payload, ...blocks];
+      const dupeBlockIndex = blocks.findIndex(
```

Comment:
Is it possible for multiple duplicate blocks?

Reply:
Yes, if you watch the logs, new blocks fire multiple events. I believe this is handled internally by the various modules within bcoin but for receiving this via websockets we need to handle it ourselves.